src/components/Pagination/pagination-utils.ts

```diff
@@ -0,0 +1,23 @@
+export function getLinkDataCy(index: number, totalPages: number): string {
+  if (index === 0) {
+    return 'prevLink';
+  } else {
+    return index === totalPages + 1 ? 'nextLink' : 'pageLink';
+  }
+}
+
+export function getLinkHref(index: number, totalPages: number): string {
+  if (index === 0) {
+    return '#prev';
+  } else {
+    return index === totalPages + 1 ? '#next' : `#${index}`;
+  }
+}
+
+export function getLinkTextContent(index: number, totalPages: number): string {
+  if (index === 0) {
+    return '«';
+  } else {
+    return index === totalPages + 1 ? '»' : `${index}`;
```

Comment:
The same indexing consideration applies here as well. Using zero-based indexing for `totalPages` may simplify the code and align with typical indexing conventions in programming.